### DISCUSSION TITLE
TMC2208 or TMC2209 for Dreamer

### DISCUSSION THREAD

**Discussion by lemmingDev** [moonglow/FlashForge_Marlin#63] [General]
Hi All

Have read we can make out FF Dreamers quieter by installing TMC drivers

Should I buy TMC2208 or TMC2209? 

The TMC2208 seem to be cheaper, and as I have 3x printers to do, I was thinking of getting 15x of these:
https://www.aliexpress.com/item/4000869320068.html

I heard we need to reverse the pin headers which can be a little time consuming, but Aliexpress doesn't seem to have any cheap ones without the pinheaders pre-soldered.

Any other advice?

**Comment by moonglow** [comment]
Hi !
>Should I buy TMC2208 or TMC2209?

I think any of them will works OK

>I heard we need to reverse the pin headers which can be a little time consuming, 
Yep you need to do that, and remove some unused pins and it will works really quiet... 

Some cons:
1. You you need to reverse pins
2. You must set motor current settings by yourself ( if you want to do that via external voltage reference you need some more changes )

Links ( in Russian with such mods ):
- With custom PCB - [LINK](https://3dtoday.ru/blogs/sobdv/tmc2208-dlya-flashforge-dreamer)
- With inverted pins - [LINK](https://3dtoday.ru/blogs/nazara/flashforge-dreamer-with-tmc2208-ili-stavim-tms2208-na-fleshika)
- Another one with inverted pins - [LINK](https://3dtoday.ru/blogs/chistota/zamena-rodnyx-draiverov-flashforge-dreamer-na-tmc2209-v20-mks)

**Reply by lemmingDev** [reply]
Thanks for the reply. I'll order some 2208 drivers
